src/methods/filterCourses.ts

```diff
@@ -135,27 +121,22 @@ const createModuleSelector = () => {
 }
 
 const applyFilter = (moduleCode: ModuleCode): void => {
-  let viewMode: "list" | "thumbnail"
-  let courses: HTMLElement[]
+  const [coursesListContainer, coursesThumbnailContainer] = [
+    document.querySelector<HTMLTableElement>(".courselist tbody"),
+    document.querySelector<HTMLElement>(".mycourses-body .section"),
+  ]
 
-  const coursesListContainer = document.querySelector(".courselist tbody")
-  const coursesThumbnailContainer = document.querySelector(
-    ".mycourses-body .section"
+  if (
+    !(coursesListContainer || coursesThumbnailContainer) ||
+    (coursesListContainer && coursesThumbnailContainer)
   )
+    return
 
-  if (coursesListContainer) {
-    viewMode = "list"
+  const coursesContainer = coursesListContainer ?? coursesThumbnailContainer
+  if (!coursesContainer) return
 
```

Comment:
@mkobayashime I think the last return expression is useless, but TSC says coursesContainer can be null. Why does it happen?

Reply:
@mkobayashime I got the answer. Thank you.